packages/eslint-plugin/src/rules/no-unused-private-class-members.ts

```diff
@@ -0,0 +1,60 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+import { createRule } from '../util';
+import { analyzeClassMemberUsage } from '../util/class-scope-analyzer/classScopeAnalyzer';
+
+type Options = [];
+export type MessageIds = 'unusedPrivateClassMember';
+
+export default createRule<Options, MessageIds>({
+  name: 'no-unused-private-class-members',
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow unused private class members',
```

Comment:
I think this is a bug and shouldn't be reported ([deploy preview playground link](https://deploy-preview-10913--typescript-eslint.netlify.app/play/#ts=5.8.2&fileType=.tsx&code=MYGwhgzhAEAqCmEAuBGaBvAUNaB6X0ATvAA4D2hS8AJtAEYCe0SDJiwhAliUtJNAFcAdgIg1s0ElwBuYKtABmZMgC5oIgLZ14haAB91AkCADcmCSQF0QnYPTCEAFAEoMEnEgAWnCADolZNAA-EEAvNAoZjgAvpixQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtYlQBuAQ3yJaZeOOTJaAW0RKARomioMkTdA7RI4MAF8QJoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)):

```ts
class Test1 {
  // reported but shouldn't?
  private foo: number | null;

  public bar() {
    this.foo ??= 1;
  }
}
```